rotkehlchen/chain/evm/accounting/interfaces.py

```diff
@@ -51,63 +60,194 @@ def reset(self) -> None:
+
+    This also handles accounting for PnL during withdrawals for the said protocols.
+
+    Examples are:
     - Curve
     - Balancer
+    - Yearn
+    - Uniswap v2 & v3
+    - Sushiswap v2
+    - Aave v1
     """
+    amounts_deposited: dict[Asset, Balance] = defaultdict(Balance)
+
+    def reset(self) -> None:
+        self.amounts_deposited: dict[Asset, Balance] = defaultdict(Balance)
 
-    def _process_deposit_or_withdrawal(
+    def process_withdrawal(
             self,
             pot: 'AccountingPot',
             event: 'EvmEvent',
             other_events: Iterator['EvmEvent'],
     ) -> None:
-        """
-        Process deposits and withdrawals from protocols that allow to deposit multiple assets
-        in return for a wrapped token. There are multiple events that we have to consume from
-        the iterator.
-
-        The receive wrapped event needs to have in the extra data field a key `deposit_events_num`
-        marking the number of events to consume.
-        The return wrapped event needs to have the key `withdrawal_events_num` with the number
-        of events in the return event.
-        """
-        method: Literal['acquisition', 'spend']
-        if event.event_type == HistoryEventType.RECEIVE:
-            # Pool token is received, which means it is a deposit
-            events_to_consume = event.extra_data.get('deposit_events_num', None) if event.extra_data is not None else None  # noqa: E501
-            method = 'spend'
-        else:  # Withdrawal
-            events_to_consume = event.extra_data.get('withdrawal_events_num', None) if event.extra_data is not None else None  # noqa: E501
-            method = 'acquisition'
-
-        if events_to_consume is None:
-            log.debug(
-                f'Could not find the number of events to consume for a {event.counterparty} '
-                f'deposit/withdrawal transaction {event.serialized_event_identifier}',
+        """Process withdrawals from protocols and deducts the USD value of the amount(s) withdrawn."""  # noqa: E501
+        if not (
+            event.extra_data is not None and
+            'withdrawal_events_num' in event.extra_data and
+            event.counterparty in {'curve', 'balancer-v1'}
+        ):
+            usd_value = self._query_historical_price(pot=pot, event=event)
+            self._calculate_pnl(
+                pot=pot,
+                event=event,
+                usd_value=usd_value,
+            )
+            return
+
+        # it's curve or balancer protocol
+        events_to_consume = event.extra_data['withdrawal_events_num']
+        for idx in range(events_to_consume):
+            next_event = next(other_events, None)
+            if next_event is None:
+                log.debug(f'Could not consume event nr. {idx} for {event.counterparty} withdrawal')  # noqa: E501
+                return
+
+            if next_event.balance.amount == ZERO:
+                continue
```

Comment:
can you log that you are skipping the event and show it? It will become handy if we have to debug anything special